native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java

```diff
@@ -0,0 +1,116 @@
+package io.ballerina.lib.sap;
+
+import com.sap.conn.jco.ext.DestinationDataEventListener;
+import com.sap.conn.jco.ext.DestinationDataProvider;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class BallerinaDestinationDataProvider implements DestinationDataProvider {
+
+    private final Map<String, Properties> destinationProperties = new HashMap<>();
+
+    @Override
+    public Properties getDestinationProperties(String destinationName) {
+        if (destinationProperties.containsKey(destinationName)) {
+            return destinationProperties.get(destinationName);
+        } else {
+            throw new RuntimeException("Destination " + destinationName + " not found");
+        }
+    }
+
+    @Override
+    public void setDestinationDataEventListener(DestinationDataEventListener eventListener) {
+    }
+
+    @Override
+    public boolean supportsEvents() {
+        return true;
+    }
+
+    public void addDestination(BMap<BString, Object> jcoDestinationConfig) {
+        Properties properties = new Properties();
+        Map<String, String> configKeys = new HashMap<>();
```

Comment:
Cant we keep this as constants map?

Reply:
Do you mean using [static <K,V> of​("k1","v1",..)](https://docs.oracle.com/javase/9/docs/api/java/util/Map.html#ofEntries-java.util.Map.Entry...-) ?